packages/plugin-spatial-nback/src/index.ts

```diff
@@ -0,0 +1,509 @@
+      const feedback_div = document.getElementById('nback-feedback') as HTMLElement;
+      const stimulus_cell = document.getElementById(`cell-${stimulus_row}-${stimulus_col}`) as HTMLElement;
+
+      // Show border feedback immediately on the grid
+      if (trial.show_feedback_border) {
+        grid.style.border = `6px solid ${is_correct ? trial.correct_color : trial.incorrect_color}`;
+      }
+
+      // Show text feedback
+      if (trial.show_feedback_text) {
+        let feedback_text = is_correct ? 'Correct!' : 'Incorrect!';
+        if (response_time !== null) {
+          feedback_text += ` (${Math.round(response_time)}ms)`;
+        }
+        feedback_div.textContent = feedback_text;
+        feedback_div.style.color = is_correct ? trial.correct_color : trial.incorrect_color;
+      }
+
+      // Handle timing based on when response occurred
+      if (made_response && !stimulus_hidden) {
+        // Response during stimulus - keep border and stimulus until stimulus duration ends
+        const elapsed_time = performance.now() - trial_start_time;
+        const remaining_stimulus_time = Math.max(0, trial.stimulus_duration - elapsed_time);
+        
+        // Wait for remaining stimulus time, then hide stimulus but keep border
+        setTimeout(() => {
+          // Hide stimulus but keep the feedback border
+          stimulus_cell.style.backgroundColor = 'white';
+          
+          // Wait for feedback duration + ISI duration before ending trial
+          setTimeout(() => {
+            endTrial(is_correct, response_time, made_response);
+          }, trial.feedback_duration + trial.isi_duration);
+        }, remaining_stimulus_time);
+        
+      } else if (made_response && stimulus_hidden) {
+        // Response during ISI - stimulus already hidden, just wait for remaining time
+        const elapsed_time = performance.now() - trial_start_time;
+        const isi_start_time = trial.stimulus_duration;
+        const elapsed_isi_time = elapsed_time - isi_start_time;
+        const remaining_isi_time = Math.max(0, trial.isi_duration - elapsed_isi_time);
+        
+        setTimeout(() => {
+          endTrial(is_correct, response_time, made_response);
+        }, remaining_isi_time + trial.feedback_duration);
+        
+      } else {
+        // No response - just wait for feedback duration
+        setTimeout(() => {
+          endTrial(is_correct, response_time, made_response);
+        }, total_feedback_duration);
+      }
+    };
+
+    const endTrial = (is_correct: boolean, response_time: number | null, made_response: boolean): void => {
+      // Prepare trial data
+      const trial_data = {
+        stimulus_row: stimulus_row,
+        stimulus_col: stimulus_col,
+        is_target: trial.is_target,
+        response: made_response,
+        response_time: response_time,
+        correct: is_correct
+      };
+
+      // Clear display
+      display_element.innerHTML = '';
```

Comment:
I think this is no longer necessary and that jsPsych does it automatically. that's a relatively recent change so my memory is fuzzy.